scripts.js

```diff
@@ -40,89 +33,89 @@ var hexCodeThree = document.querySelector('.hex-code-3');
 var hexCodeFour = document.querySelector('.hex-code-4');
 var hexCodeFive = document.querySelector('.hex-code-5');
 
+// var palette = document.querySelector('.palette');
+// var colorProps = document.querySelectorAll('.color-props');
 var savedPaletteSection = document.querySelector('.saved-section');
-var colorProps = document.querySelectorAll('.color-props');
 
-window.addEventListener('load', createPalette);
-// window.addEventListener('load', generateColors);
+window.addEventListener('load', makeNewPalette);
 
-newPaletteButton.addEventListener('click', generateColors);
+newPaletteButton.addEventListener('click', makeNewPalette);
 savePaletteButton.addEventListener('click', savePalette);
-// lockButton.addEventListener('click', lockColor);
+
+savedPaletteSection.addEventListener('click', function(event) {
+   if (event.target.className === 'trash-icon') {
+     for (var i = 0; i < savedPalettes.length; i++) {
+       if (`${savedPalettes[i].id}` === event.target.id) {
+         savedPalettes.splice(i, 1);
+       }
+     }
+   }
+  displaySavedPalettes();
+ });
 
 colorOneDiv.addEventListener('click', function(event) {
   if (event.target.className === 'lock-color-1' || 'unlock-color-1') {
-    lockButtonOne.classList.toggle('hidden')
-    unlockButtonOne.classList.toggle('hidden')
+    lockButtonOne.classList.toggle('hidden');
+    unlockButtonOne.classList.toggle('hidden');
   }
+
   if (`${lockButtonOne.classList}` === 'lock-color-1') {
     color1.locked = true;
   } else {
-      color1.locked = false;
+    color1.locked = false;
   }
 });
```

Comment:
Trying to write DRY code should always be your goal, but there's a constant tradeoff between time/effort and rewriting a little bit of code. You have to decide if it's worth the time to refactor or leave something as is, and that intuition takes a while to develop (no pun intended). That said I think this is a good opportunity to do so. 

Because of the nature of the `colorBlankDiv` event listeners here, I think we can refactor this to be a lot cleaner. In order to build something dynamic you have to consider what's different between each case. Here I really only see three things different (once the first if comparison is removed from my previous comment): `lockButtonOne, unlockButtonOne, color1`. So, we can build a function to take these in three parameters: `lockButton`, `unlockButton`, `color`. After refactoring you might end up with something like this:
<img width="477" alt="Screen Shot 2022-02-28 at 9 37 55 AM" src="https://user-images.githubusercontent.com/69563078/156021867-79b3a8cd-e05b-47f0-bd1b-e37839c7c946.png">
 
 note: I used `classList.contains('hidden')` here. I think it's a little bit more readable because as I'm looking at this code I don't know the significance of if the classList is `lock-color-1`. I have to dive into the html to discover that the idea here is that if that's the only thing in the classList then the lock is visible because 'hidden' isn't in the classList so the user locked it so we should lock that color in the palette. When I see `if (!lockButton.classList.contains('hidden')` my brain goes "okay, if the lock button is visible..."
 
This is a very small thing ultimately, but my job here is to press you on this stuff so I'm just trying to expose you to it. :)